src/software/simulated_tests/ball_speed_threshold.py

```diff
@@ -35,8 +38,82 @@ def get_validation_status(self, world) -> ValidationStatus:
 
     def get_validation_geometry(self, world) -> ValidationGeometry:
         """override"""
-        # TODO (#2556): visualize
-        return create_validation_geometry([])
+        ball_speed = math.sqrt(
```

Comment:
this function is common w/ the robot speed one and is fairly similar to the one in thunderscope

it would be good to put these functions in `simulated_tests/speed_threshold.py` and have ball_speed and robot_speed use its functions.

It doesn't have to implement anything itself, you can just say speed_threshold helper functions